### ISSUE TITLE
Double slash makes the fromat incompatible with uri standard

### ISSUE THREAD

**Issue by mpazik** [hash-uri/hash-uri#6]
According to [Section 3 of RFC-3986](https://tools.ietf.org/html/rfc3986#section-3)  
> When  authority is not present, the path cannot begin with two slash characters ("//")

I belive the who slashes could be omitted and as the result the hash uri would look like
`hash:sha256/9f86d081884c7d659a2feaa0?type=text/plain#top`

other option is to follow it with a single hash that would stuggest root directory
`hash:/sha256/9f86d081884c7d659a2feaa0?type=text/plain#top`

or three slashes like in the case of ni scheme [RFC-3986](https://tools.ietf.org/html/rfc6920) which would mean that authroity is empty. But I don't think it makes sense as athority can not be ever present in the hash uri
`hash:///sha256/9f86d081884c7d659a2feaa0?type=text/plain#top`

**Comment by btrask** [comment]
Hi @mpazik. Thanks for taking the time to comment. Under this specification, the algorithm name (e.g. "sha256") is considered the authority. Thus using the double slashes is correct.

It also has the practical advantages of being more recognized, easier to parse, etc.

**Comment by milahu** [comment]
> advantages of being more recognized

not really. `//name` is usually a hostname like `//localhost` or `//google.com`

also you waste 2 chars

> other option is to follow it with a single hash that would suggest root directory

this. some day we want to implement this in the context of the linux FHS
[my suggestion](https://discourse.nixos.org/t/nix-sha256-is-bug-not-feature-solution-a-global-cas-filesystem/15791) was `/cas` (content addressed store) but `/hash` is prettier

so for example, the url x would be mounted as y:

```
 hash:git/commit/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/tree/path/to/file.txt
/hash/git/commit/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/tree/path/to/file.txt
```
